browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,263 @@
+  auto* widget = new views::Widget();
+  widget->Init(std::move(params));
+
+  location_bar->view_observation_.Observe(web_view);
+  location_bar->UpdateVisibility();
+  location_bar->UpdateBounds();
+
+  widget->Show();
+
+  return location_bar;
+}
+
+void SplitViewLocationBar::SetWebContents(content::WebContents* web_contents) {
+  if (this->web_contents() == web_contents) {
+    return;
+  }
+
+  Observe(web_contents);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnPaintBackground(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ true);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorToolbar));
+  flags.setAntiAlias(true);
+  canvas->DrawPath(path, flags);
+}
+
+void SplitViewLocationBar::OnPaintBorder(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ false);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+gfx::Size SplitViewLocationBar::CalculatePreferredSize(
+    const views::SizeBounds& available_size) const {
+  auto size = views::WidgetDelegateView::CalculatePreferredSize(available_size);
+  if (auto* view = view_observation_.GetSource(); view && view->width()) {
+    size.set_width(std::min(view->width(), size.width()));
+  }
+  return size;
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::DidChangeVisibleSecurityState() {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    auto url = contents->GetVisibleURL();
+    auto origin = url::Origin::Create(url);
+    auto formatted_origin = url_formatter::FormatUrl(
```

Comment:
on second thought, why not just show the full origin here? i'm not sure what happens if the url is `blob:` or `view-source:` but it seems important to include the scheme

Reply:
Sure. It'll be like

<img width="583" alt="image" src="https://github.com/brave/brave-core/assets/5474642/5301a053-fa08-4a00-9e62-f40c1762c51d">

<img width="592" alt="image" src="https://github.com/brave/brave-core/assets/5474642/818c97ec-538a-4c73-945b-2a29f972eb3f">

<img width="592" alt="image" src="https://github.com/brave/brave-core/assets/5474642/3cfb2bf6-1d3e-4879-99d4-fbc35096e46c">

I also couldn't find nice way to iterate "safe" schemes. Do you think we can omit the icon then?

Reply:
I found that work for us in that case

```
  auto security_level = SecurityStateTabHelper::FromWebContents(web_contents())
                            ->GetSecurityLevel();
  return security_level == security_state::SecurityLevel::SECURE ||
         security_level == security_state::SecurityLevel::NONE;
```

Reply:
IMO it is more informative to omit the icon and just show the full origin (preferably with a slash through HTTPS on invalid certs)

Reply:
Probably better to show the scheme, only if it's not `https` here

Reply:
So, do we want it to be like
* Hide scheme when the security level is secure - https:// and valid cert -> example.com
* Show slashed when it's https:// but not valid - ~~https~~://example.com
* Show scheme when the security level is  none(neutral, like chrome:// view-source) - chrome://newtab

And remove the icon entirely. Am I correct?

Reply:
IMO it'd be confusing to only hide scheme if HTTPS. i understand it takes up more space though.

otherwise sgtm

Reply:
@diracdeltas it's the same behavior of the omnibar. I feel being not consistent is confusing.
Also it looks bad visually, but that's secondary.

Reply:
I would leave the icon for not secure (none // not secure, and when we strikethrough the ~~https://~~

Reply:
As yan mentioned, there is a config to show the full URL (with the schema), all the time. We should probably follow that configuration and show `https` all the times, in that particular case.

I don't see much point in having drift from the omnibar behaviour

Reply:
it turns out i have the `Always show full URLs` setting enabled whereas @thypon does not. in any case, we agree it'd be least confusing if this mini-URL view just showed the same as whatever the user would see in the omnibox minus the URL path (so following that setting)

Reply:
yes, also fine with me to show the icon as long as it matches what is shown in the omnibox 100% of the time. whatever chromium does to determin the omnibox icon, we should just copy that logic.

Reply:
Thanks all, I'll apply the logic above only when "Always show full URLs" is off. And leave icon as is.